pybossa/view/projects.py

```diff
@@ -2516,7 +2516,7 @@ def respond():
 
         reserve_category_columns = form.reserve_category_columns.data if form.sched.data in ["task_queue_scheduler"] else []
         project.info["reserve_tasks"] = dict(category=reserve_category_columns)
-        project.info['tasklist_columns'] = form.customized_columns.data
+        project.info['tasklist_columns'] = form.customized_columns.data or project.info.get('tasklist_columns', [])
```

Comment:
I guess this would be problematic - if user intends to remove `tasklist_columns`, this code would ignore the behavior and keep the old data.

perhaps we can test this code on https://qa.gigwork.net/project/testtaskqueue/tasks/scheduler as well?

Reply:
Yes, this is correct and confirmed.

This fixes the more urgent issue of data being lost, but will not allow the user to then clear ALL columns, they would have to save at least 1 column.

Reply:
It seems both pages call the same API POST /tasks/scheduler with a payload containing `customized_columns=[]`, with no way to differentiate an intentional emptying of values by the user from a default one via the wizard bar.

A fix for this could be to include the customized columns UI in the wizard settings so that the data saves correctly OR to send a dummy value for customized_columns when the wizard saves and ignore this value on the server.

Reply:
> A fix for this could be to include the customized columns UI in the wizard settings so that the data saves correctly OR to send a dummy value for customized_columns when the wizard saves and ignore this value on the server.

Agree, we do have a ticket to add customized columns to wizard settings in backlog.

Reply:
Fix for this case added https://github.com/bloomberg/pybossa-default-theme/pull/369
This PR may be closed without merge.